config/common/common.go

```diff
@@ -95,3 +97,18 @@ func TranslateReportPaths(r *report.Report, ts translate.TranslationSet) {
 		r.Entries[i].Context = context
 	}
 }
+
+// ValidateTranslated validates the final translated JSON config.
+func ValidateTranslated(r *report.Report, translated interface{}, contextTree tree.Node, ts translate.TranslationSet) {
+	// Check for invalid duplicated keys.
+	dupsReport := validate.ValidateCustom(translated, "json", ignvalidate.ValidateDups)
```

Comment:
Self note: Ignition's `ValidateWithContext` utility function (which `ValidateTranslated` is partially reimplementing) calls `ValidateDups` _after_ `validate.Validate` but it probably doesn't matter.

Reply:
There is possibly a larger rework to be done on Ignition side (I'm not sure why `ValidateWithContext` sidesteps the normal validation entrypoint), but I honestly got a bit lost in validation indirection layers trying to figure out where to best plug it.